Appendix/stemming (#57)
MrLongbottom/P10-Article: sections/appendix/stemming.tex

\subsection{Stemming the dataset}\label{sec:stemming}

> In this project we have done minimal preprocessing of the dataset, because in a previous project, a more aggressive preprocessing step turned out to have a negative effect on the \gls{lda} model.

Reviewer: ```suggestion
In this project, we have done minimal preprocessing of the dataset, because, in a previous project, a more aggressive preprocessing step turned out to hurt the performance of the \gls{lda} model.
```

Though, after further consideration, it is worth looking into what effect adding just stemming to our current preprocessing, described in \autoref{sec:dataset}, would have on the standard \gls{lda} model.

With stemming included, the number of unique words goes down to $60,651$ from the previous $69,192$ unique words.
This means that 8541 unique words have been stemmed down to their root forms.
Even though much fewer unique words are in our dataset, which can have an effect on which articles are removed, the dataset only goes from $139,064$ articles to $139,060$, which is a removal of 4 articles.

In the non-stemmed \gls{lda} model, words that have no meaning topically seem to have a large influence.
This is still the case in the stemmed model, where words like 'du' (you), 'mig' (me), and 'a' still appear in the top words of topics.
To remove these words, we would have to include some more advanced preprocessing in the form of stop word removal and part-of-speech (POS) tagging.

Examples of top words for similar topics between the models can be seen in \autoref{tab:topic_examples}.
It is seen here that some of the topics between the models are clearly about the same subjects.

> The stemmed model's topics might also be slightly more understandable, because words with the same meaning (e.g. 'politi' and 'politiet') do not appear.

Reviewer: ```suggestion
The stemmed model's topics might also be slightly more understandable because words with the same meaning (e.g. 'politi' and 'politiet') do not appear.
```

> Other than this, including stemming does not seem to have made a significant difference.

Reviewer: ```suggestion
Other than this, including stemming, does not seem to have made a significant difference.
```
